lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/VortexManager.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.vortex.driver;
+
+import org.apache.reef.tang.annotations.DefaultImplementation;
+import org.apache.reef.vortex.driver.tasklet.Tasklet;
+import org.apache.reef.vortex.driver.tasklet.TaskletExecutor;
+
+import java.io.Serializable;
+
+/**
+ * The heart of Vortex.
+ * Processes various tasklet related events coming from different components of the system.
+ */
+@DefaultImplementation(DefaultVortexManager.class)
+public interface VortexManager {
```

Comment:
Please refer to the comments I left at the `VortexManager` implementation. It'd be great to crystallize `VortexManager`, `VortexFunction`, `Tasklet`, `VortexUserCode`, `VortexThreadPool`, etc.